src/main/java/org/junit/ExceptionSupplier.java

```diff
@@ -0,0 +1,13 @@
+package org.junit;
+
+/**
+ * A functional interface used to call code that is expected to throw an exception
+ * @param <T> The type of exception expected to be thrown by {@link ExceptionSupplier#get()}
+ */
+public interface ExceptionSupplier<T extends Exception> {
+    /**
+     * Execute code that is expected to throw an exception
+     * @throws T the type of exception expected to be thrown
+     */
+    void get() throws T, Exception;
```

Comment:
It has to throw `Exception` just in case it calls a method that declares a checked exception but isn't expected to throw it. Strictly speaking, the `T` isn't necessary, but having it declares the semantic expectations much better. If the method didn't throw `T`, then the interface would have a type parameter that it doesn't use — and the type parameter really helps explain what's going to happen when calling the expect method.